### ISSUE TITLE
"Rust Analyzer: Run" command should be available in Cargo.toml files

### ISSUE THREAD

**Issue by matklad** [rust-lang/rust-analyzer#13291]
This is implemented here:

https://github.com/rust-lang/rust-analyzer/blob/73ab709b38b171d561c119f5c6f94af1bf2e4f3b/crates/rust-analyzer/src/handlers.rs#L690

https://github.com/rust-lang/rust-analyzer/blob/73ab709b38b171d561c119f5c6f94af1bf2e4f3b/editors/code/src/commands.ts#L835

**Comment by MonliH** [comment]
I've never contributed to RA before, but I'd like to give this a stab.

**Comment by MonliH** [comment]
Hmm... One thing I'm running into right now is that getting the `CargoTargetSpec::for_file` requires some file_id. However, as far as I know, the `Cargo.toml` file is not in the `GlobalStateSnapshot.vfs` and so `from_proto::file_id` fails: https://github.com/rust-lang/rust-analyzer/blob/73ab709b38b171d561c119f5c6f94af1bf2e4f3b/crates/rust-analyzer/src/handlers.rs#L695

Should we put the `Cargo.toml` file in the `GlobalStateSnapshot.vfs`? If so, where is the best place to do so?

**Comment by matklad** [comment]
Yeah, the "first best" case here, where we suggest based on what target in Cargo.toml the cursor at, is going to be a stack of yaks! But I think "first worst" already is an improvement:

* in workspace, suggest `cargo check -p this-one` based on which cargo.toml is open
* suggest `cargo run -p this-one --bin foo` based on what `cargo metadata` tells us about binaries.
